pkg/kp/lock_test.go

```diff
@@ -0,0 +1,39 @@
+package kp
+
+import (
+	"testing"
+	"time"
+)
+
+const (
+	lockMessage = "Locked by lock_test.go"
+)
+
+func TestLock(t *testing.T) {
```

Comment:
I want this test to be clearer about what it's verifying. From reading the test I can tell that it's establishing a lock and showing that a client that does not use the lock cannot access a key. However, I don't see tests that show that owning the lock does allow reads and writes.